src/app/layout.tsx

```diff
@@ -15,7 +20,19 @@ export default function RootLayout({
   return (
     <html lang="en">
       <body>
-        <ClientProviders>{children}</ClientProviders>
+        <Grid className="grid min-h-screen grid-rows-[auto,1fr,auto]">
+          <div className="row-start-1 row-end-2">
+            <Header />
+          </div>
+          <div className="row-start-2 row-end-3 overflow-auto">
+            <div className="mx-auto max-w-7xl px-4 py-8">
+              <ClientProviders>{children}</ClientProviders>
```

Comment:
can we move this back to be direct child of <body>?

Something like:
```
   <html lang="en">
      <body>
        <ClientProviders>
          <Grid className="grid min-h-screen grid-rows-[auto,1fr,auto]">
            <div className="row-start-1 row-end-2">
              <Header />
            </div>
            <div className="row-start-2 row-end-3 overflow-auto">
              <div className="mx-auto max-w-7xl px-4 py-8">{children}</div>
            </div>
            <div className="row-start-3 row-end-4">
              <Footer />
            </div>
          </Grid>
        </ClientProviders>
      </body>
    </html>
```

Those providers almost never render any DOM nodes, so their position doesn't impact the UI, but they are usually very near to the top of the components tree because you want anything under them to be able to access whatever they provide